File: 228_SummaryRanges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) 
{
    let result = [];
    let start = 0;
    let flag = true;
    for(let i = 0; i < nums.length; i++)
    {
        if(flag) 
        {
            start = nums[i];
            flag = false;
        }
        if(nums[i]+1 != nums[i+1]) {
            let end = nums[i];
            if(start == end)
            {
                result.push(String(start));
            } else {
                result.push(String(start+"->"+end))
            }
            flag = true;
        }
    }   
    return result; 
};

let main = function()
{
    
}

File: 228_SummaryRanges.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution 
{
    public:
        vector<string> summaryRanges(vector<int>& nums) 
        {
            vector<string> result;
            int n = nums.size();
            if (n == 0) return result;

            int start = nums[0];

            for (int i = 0; i < n; i++) {
                if (i == n - 1 || nums[i] + 1 != nums[i + 1]) {
                    int end = nums[i];

                    if (start == end) {
                        result.push_back(to_string(start));
                    } else {
                        result.push_back(to_string(start) + "->" + to_string(end));
                    }
                    if (i + 1 < n) {
                        start = nums[i + 1];
                    }
                }
            }
            return result;
        }
};


int main()
{

    return 0;
}